amps/ags/rm-automation/rm-automation-community-rest-api/src/main/java/org/alfresco/rest/rm/community/requests/gscore/api/HoldsAPI.java

```diff
@@ -0,0 +1,160 @@
+ * provided under the following open source license terms:
+ * -
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * -
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * -
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.rest.rm.community.requests.gscore.api;
+
+import static org.alfresco.rest.core.RestRequest.requestWithBody;
+import static org.alfresco.rest.core.RestRequest.simpleRequest;
+import static org.alfresco.rest.rm.community.util.ParameterCheck.mandatoryObject;
+import static org.alfresco.rest.rm.community.util.ParameterCheck.mandatoryString;
+import static org.alfresco.rest.rm.community.util.PojoUtility.toJson;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.springframework.http.HttpMethod.DELETE;
+import static org.springframework.http.HttpMethod.GET;
+import static org.springframework.http.HttpMethod.POST;
+import static org.springframework.http.HttpMethod.PUT;
+
+import org.alfresco.rest.core.RMRestWrapper;
+import org.alfresco.rest.rm.community.model.hold.Hold;
+import org.alfresco.rest.rm.community.model.hold.HoldChild;
+import org.alfresco.rest.rm.community.model.hold.HoldChildCollection;
+import org.alfresco.rest.rm.community.model.hold.HoldDeletionReason;
+import org.alfresco.rest.rm.community.model.hold.HoldDeletionReasonEntry;
+import org.alfresco.rest.rm.community.requests.RMModelRequest;
+
+public class HoldsAPI extends RMModelRequest
+{
+
+    /**
+     * @param rmRestWrapper
+     */
+    public HoldsAPI(RMRestWrapper rmRestWrapper)
+    {
+        super(rmRestWrapper);
+    }
+
+
+    public Hold getHold(String holdId)
+    {
+        mandatoryString("holdId", holdId);
+
+        return getHold(holdId, EMPTY);
+    }
+
+    public Hold getHold(String holdId, String parameters)
+    {
+        mandatoryString("holdId", holdId);
+
+        return getRmRestWrapper().processModel(Hold.class, simpleRequest(
+            GET,
+            "holds/{holdId}?{parameters}",
+            holdId,
+            parameters
+                                                                        ));
+    }
+
+    public Hold updateHold(Hold holdModel, String holdId)
+    {
+        mandatoryObject("holdModel", holdModel);
+        mandatoryString("holdId", holdId);
+
+        return updateHold(holdModel, holdId, EMPTY);
+    }
+
+    public Hold updateHold(Hold holdModel, String holdId, String parameters)
```

Comment:
I don't think this is used in the tests yet.

Reply:
I think the same is true of some of the other REST methods.

Reply:
Right, this is not used in the tests yet. I'm going to use that in the next PR to add more integration tests.